app/controllers/api/v1/steps_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class StepsController < BaseController
+      before_action :load_team, :load_project, :load_experiment, :load_task, :load_protocol
+      before_action only: :show do
+        load_step(:id)
+      end
+
+      def index
+        steps = @protocol.steps
+                         .page(params.dig(:page, :number))
+                         .per(params.dig(:page, :size))
+
+        render jsonapi: steps, each_serializer: StepSerializer
+      end
+
+      def show
+        render jsonapi: @step, serializer: StepSerializer
+      end
+
+      def create
+        raise PermissionError.new(Protocol, :create) unless can_manage_protocol_in_module?(@protocol)
+
+        step = @protocol.steps.create!(step_params.merge!(completed: false,
```

Comment:
We do not have specified requirements. Because the user does not have the ability to create completed step over UI on the web app, I use the same principle here. But maybe I am wrong...?

Reply:
I see, good point.